ui/src/map/layers/alignment/location-track-duplicate-endpoint-indicator-layer.ts

```diff
@@ -0,0 +1,234 @@
+            pixelRatio * DASHED_LINE_INDICATOR_FONT_SIZE
+        }px ${mapStyles['alignmentBadge-font-family']}`;
+
+        ctx.save();
+        const textPositionOffset =
+            endpointType === 'START' ? 0 : 2 * indicatorTextBackgroundHeight(pixelRatio);
+
+        const nameText = name;
+        const nameTextWidth = ctx.measureText(nameText).width;
+        const nameTextXEndPosition =
+            x - indicatorLineWidth(pixelRatio) * (positiveXOffset ? 1 : -1);
+        const nameTextYEndPosition = y - indicatorLineHeight(pixelRatio) + textPositionOffset;
+
+        const trackMeterText = trackMeter ? formatTrackMeter(trackMeter) : '';
+        const trackMeterTextWidth = ctx.measureText(trackMeterText).width;
+        const trackMeterTextXEndPosition =
+            x - indicatorLineWidth(pixelRatio) * (positiveXOffset ? 1 : -1);
+        const trackMeterTextYEndPosition =
+            y -
+            indicatorLineHeight(pixelRatio) +
+            indicatorTextBackgroundHeight(pixelRatio) +
+            textPositionOffset;
+
+        ctx.translate(x, y);
+        ctx.rotate(rotation);
+        ctx.translate(-x, -y);
+
+        ctx.fillStyle = 'rgba(255, 255, 255, 0.85)';
+        ctx.fillRect(
+            nameTextXEndPosition,
+            endpointType === 'START' ? y - indicatorTextBackgroundHeight(pixelRatio) * 2 : y,
+            nameTextWidth + indicatorTextPadding(pixelRatio) * 2,
+            indicatorTextBackgroundHeight(pixelRatio),
+        );
+        ctx.fillRect(
+            trackMeterTextXEndPosition,
+            endpointType === 'START'
+                ? y - indicatorTextBackgroundHeight(pixelRatio)
+                : y + indicatorTextBackgroundHeight(pixelRatio),
+            trackMeterTextWidth + indicatorTextPadding(pixelRatio) * 2,
+            indicatorTextBackgroundHeight(pixelRatio),
+        );
+
+        //Dashed line
+        ctx.beginPath();
+        ctx.lineWidth = indicatorDashedLineWidth(pixelRatio);
+        ctx.strokeStyle = BLACK;
+        ctx.setLineDash(indicatorLineDash);
+        ctx.moveTo(x, y);
+        ctx.lineTo(nameTextXEndPosition, y);
+        ctx.stroke();
+
+        ctx.fillStyle = BLACK;
+        ctx.textAlign = positiveXOffset ? 'left' : 'right';
+        ctx.textBaseline = 'bottom';
+        ctx.fillText(
+            nameText,
+            nameTextXEndPosition - indicatorTextPadding(pixelRatio) * (positiveXOffset ? -1 : 1),
+            nameTextYEndPosition,
+        );
+        ctx.fillText(
+            trackMeterText,
+            trackMeterTextXEndPosition -
+                indicatorTextPadding(pixelRatio) * (positiveXOffset ? -1 : 1),
+            trackMeterTextYEndPosition,
+        );
+
+        ctx.restore();
+    };
+
+    feature.setStyle(() => new Style({ renderer }));
+    return feature;
+}
+
+const calculateZIndexForTrackMeter = (trackMeter: TrackMeter): number =>
+    Math.floor(parseInt(trackMeter.kmNumber.substring(0, 4)) * 10000 + trackMeter.meters);
```

Comment:
Härpäkkeet halutaan piirtää niin, että tilanteissa joissa ne menisivät päällekäin, pidemmällä rataa olevat piirtyvät aina aiempien päälle. Täten inferroidaan z-indeksi suoraan kunkin härpäkkeen kohdan rataosoitteesta